crates/router/src/services/authentication.rs

```diff
@@ -2504,17 +2545,25 @@ where
     T: serde::de::DeserializeOwned,
     A: SessionStateInfo + Sync,
 {
-    let token = match get_cookie_from_header(headers).and_then(cookies::parse_cookie) {
-        Ok(cookies) => cookies,
-        Err(error) => {
-            let token = get_jwt_from_authorization_header(headers);
-            if token.is_err() {
-                logger::error!(?error);
-            }
-            token?.to_owned()
-        }
+    let cookie_token_result = get_cookie_from_header(headers).and_then(cookies::parse_cookie);
+    let auth_header_token_result = get_jwt_from_authorization_header(headers);
+    let use_cookie_only = state.conf().user.force_cookies;
+
+    logger::info!(
+        user_agent = ?headers.get(headers::USER_AGENT),
+        header_names = ?headers.keys().collect::<Vec<_>>(),
+        is_token_equal =
+            auth_header_token_result.as_deref().ok() == cookie_token_result.as_deref().ok(),
+        use_cookie_only,
+    );
+
+    let final_token = if use_cookie_only {
+        cookie_token_result?
+    } else {
+        cookie_token_result.unwrap_or(auth_header_token_result?.to_owned())
```

Comment:
why are we using cookie when config is false? Is this the desired behaviour?

Reply:
Nope, will change this to use auth header.